### PR TITLE
nbmodel integration - sync text source

### PR DESCRIPTION
**Pull request by dmonad**

### CODE REVIEW

File: packages/codeeditor/src/editor.ts [hunk 1]
```diff
@@ -218,21 +220,74 @@ export namespace CodeEditor {
...
+            this.nbmodel.modifySource(event.start, event.end - event.start);
+            break;
+          default:
+            this.nbmodel.setSource(value.text);
+            break;
+        }
+      });
+    }
+
+    readonly nbmodel = nbmodel.StandaloneCellFactory.createCodeCell();
```
Comment (echarles) [inline packages/codeeditor/src/editor.ts]: maybe rename nbmodel to nbcell?

File: packages/nbmodel/src/api.ts [hunk 2]
```diff
@@ -81,6 +81,10 @@ export interface ISharedBaseCell<Metadata extends nbformat.IBaseCellMetadata>
   extends IDisposable {
   getSource(): string;
   setSource(value: string): void;
+  /**
+   * Replace content from `start' to `end` with `value`.
+   */
+  modifySource(start: number, end: number, value?: string): void;
```
Comment (echarles) [inline packages/nbmodel/src/api.ts]: maybe updateSource?

Comment (echarles) [reply packages/nbmodel/src/api.ts]: or changeSource (as we discussed...)?

Comment (echarles) [reply packages/nbmodel/src/api.ts]: btw other candidate is mutateSource - But it should the same name as the existing method to update/change/mutate the metadata.

**Comment by dmonad** [comment]
I implemented the suggestions. This is ready to be merged.

**Review by echarles — APPROVED** [review]
LGTM Thx @dmonad

**Pull request merged**